Cargo-minimal.lock

```diff
@@ -235,109 +238,102 @@ dependencies = [
 
 [[package]]
 name = "itoa"
-version = "0.4.3"
+version = "0.4.8"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "1306f3464951f30e30d12373d31c79fbd52d236e5e896fd92f96ec7babbbe60b"
+checksum = "b71991ff56294aa922b450139ee08b3bfc70982c6b2c7562771375cf73542dd4"
 
 [[package]]
 name = "lazy_static"
-version = "1.4.0"
+version = "1.5.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "e2abad23fbc42b3700f2f279844dc832adb2b2eb069b2df918f455c4e18cc646"
+checksum = "bbd2bcb4c963f2ddae06a2efc7e9f3591312473c50c6685e1f298068316e66fe"
 
 [[package]]
 name = "libc"
-version = "0.2.155"
+version = "0.2.172"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "97b3888a4aecf77e811145cadf6eef5901f4782c53886191b2f693f24761847c"
+checksum = "d750af042f7ef4f724306de029d18836c26c1765a54a6a3f094cbd23a7267ffa"
 
 [[package]]
 name = "memmap2"
-version = "0.9.4"
+version = "0.9.5"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "fe751422e4a8caa417e13c3ea66452215d7d63e19e604f4980461212f3ae1322"
+checksum = "fd3f7eed9d3848f8b98834af67102b720745c4ec028fcd0aa0239277e7de374f"
 dependencies = [
  "libc",
 ]
 
 [[package]]
 name = "ppv-lite86"
-version = "0.2.8"
+version = "0.2.21"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "237a5ed80e274dbc66f86bd59c1e25edc039660be53194b5fe0a482e0f2612ea"
+checksum = "85eae3c4ed2f50dcfe72643da4befc30deadb458a9b590d720cde2f2b1e97da9"
+dependencies = [
+ "zerocopy",
+]
 
 [[package]]
 name = "proc-macro2"
-version = "1.0.63"
+version = "1.0.94"
```

Comment:
In 51fd4e86b51a199b8553e6f32295b0ab9c042307:

When I run the command locally I get 1.0.95, even with the same nightly version as used by CI. I worry that the `-Z direct-minimal-versions` flag is unstable somehow.

Reply:
I think in this case, there was a [patch release](https://crates.io/crates/proc-macro2/versions) for the `proc-macro2` crate in the time since I generated the lockfile.

Reply:
Oh, lol, ok.

But then the question becomes -- why does cargo update want to use this version? In general is it going to keep bumping versions of non-direct dependencies like this?

Reply:
I think this is an example of the crux of the trade off with `direct-minimal-versions`. It uses the minimal version for any direct dep, which I am pretty sure in a workspace just means anything in the [dependencies] section of each crate's manifest, but for transitive dependencies it uses the default policy for the constraint. Usually "update to highest minor and patch". The idea being that they might not have minimal versions specified correctly, so not testing *their* constraints. Looks like `proc-macro2` is a transitive dep?

Reply:
Ok, we definitely do not want "update to highest minor and patch". This will result in dependency choices that don't even compile on our MSRV.

Reply:
Thanks for all the work you put into this! But it seems like this cargo flag is yet another "do something weirdly specifically wrong" flag.

Reply:
Oof, good point. I'll put this back into draft and see if I can salvage anything from it, but seems like a big hole. I'll run an experiment or two with the new v3 resolver which respects MSRV settings.

Fwiw it looks like all transitive dep versions work with the MSRV which is nice. But agree that it would be a confusing failure when that breaking upgrade arrived.